workspaces/liferay-ticket-workspace/client-extensions/liferay-ticket-custom-element/src/pages/TicketsOverview.tsx

```diff
@@ -0,0 +1,170 @@
+/* eslint-disable quote-props */
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import ClayButton from '@clayui/button';
+import {Option, Picker} from '@clayui/core';
+import ClayLayout from '@clayui/layout';
+import {ClayPaginationBarWithBasicItems} from '@clayui/pagination-bar';
+import {useContext, useState} from 'react';
+import {QueryClient, useMutation} from 'react-query';
+
+import {RecentActivity} from '../components/RecentActivity';
+import {TicketGrid} from '../components/TicketGrid';
+import {QueryClientContext} from '../context';
+import {useRecentTickets} from '../hooks/useRecentTickets';
+import {useTickets} from '../hooks/useTickets';
+import {Liferay} from '../services/liferay';
+import {generateNewTicket} from '../services/tickets';
+import useDebounce from '../services/useDebounce';
+import {Filter} from '../types';
+
+const initialFilterState: Filter = {
+	field: '',
+	label: '',
+	value: '',
+};
+
+const filters: Filter[] = [
+	{
+		field: '',
+		label: 'No Filter',
+		value: '',
+	},
+	{
+		field: 'ticketStatus',
+		label: 'Open Issues',
+		value: 'open',
+	},
+	{
+		field: 'ticketStatus',
+		label: 'Queued Issues',
+		value: 'queued',
+	},
+	{
+		field: 'priority',
+		label: 'Major Priority Issues',
+		value: 'major',
+	},
+	{
+		field: 'resolution',
+		label: 'Unresolved Issues',
+		value: 'unresolved',
+	},
+];
+
+const TicketsOverview: React.FC = () => {
+	const queryClient: QueryClient = useContext(QueryClientContext);
+
+	const [filter, setFilter] = useState(initialFilterState);
+	const [page, setPage] = useState(1);
+	const [pageSize, setPageSize] = useState(20);
+	const [searchToDebounce, setSearchToDebounce] = useState<string>('');
+	const search = useDebounce(searchToDebounce, 500);
```

Comment:
This naming is a little confusing... it would make more sense to do it like this:

```
	const [search, setSearch] = useState<string>('');

	const debouncedSearch = useDebounce(search, 300);
```
We need a line break between the useState and useDebounce because they are different and debouncedSearch depends on search.

Also our default debounce time is usually `300`